lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -30,6 +60,42 @@ class LabelAndValue extends RoleManager {
 
   @override
   void update() {
+    final String? computedLabel = _computeLabel();
+
+    if (computedLabel == null) {
+      _cleanUpDom();
+      return;
+    }
+
+    _updateLabel(computedLabel);
+  }
+
+  DomElement? _labelSpan;
+
+  void _updateLabel(String label) {
+    owner.setAttribute('aria-label', label);
+
+    final bool needsDomText = !semanticsObject.hasChildren && userAgentNeedsDomText;
+
+    if (needsDomText) {
+      DomElement? span = _labelSpan;
+
+      // Lazy-create and cache the span.
+      if (span == null) {
+        _labelSpan = span = createDomElement('span');
+        semanticsObject.element.appendChild(span);
+      }
+
+      span.innerText = label;
```

Comment:
Hmm, tried it, but doesn't fully work. The local variable is needed so that the analyzer can prove that it's not null after initialization. `_labelSpan` is nullable and therefore requires a null assertion (line `_labelSpan.innerText = label;` does not compile). Also, `appendChild` return `DomNode` instead of `DomElement`, which requires an extra dynamic cast. Keeping the existing variant for now.